docs/scos/dev/upgradability-services/new-documentation/core-method-used-on-project-level.md

```diff
@@ -0,0 +1,83 @@
+---
+title: A core method is used on the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+## A core method is used on the project level
+
+Modules have public and private APIs. The public API includes the entities like facade, plugin stack, dependency list, etc. This check covers private API entities, including Business model, Factory, Dependency provider, Repository, and Entity manager.
+
+While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility in minor releases is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during minor updates.
+
+#### Solution to resolve the issue
+
+To solve this problem with unexpected issues after updating the core's private API, replace the core methods with their copies on the project level.
```

Comment:
Please add:
"Meanwhile we are working on introducing a way to report such cases and add more extension points in the core"

Reply:
Add as inform block for pages when exist internal API